packages/typespec-client-generator-core/src/internal-utils.ts

```diff
@@ -530,6 +533,30 @@ export function getHttpOperationResponseHeaders(
   return headers;
 }
 
+export function filterApiVersionsInEnum(
```

Comment:
it seems this func has many duplicate code with `getDefaultApiVersion`, can we do soem consolidation?